server/src/main/scala/org/http4s/server/middleware/CSRF.scala

```diff
@@ -367,6 +368,7 @@ object CSRF {
       httpOnly: Boolean,
       domain: Option[String] = None,
       path: Option[String] = None,
+      sameSite: SameSite = SameSite.Lax,
```

Comment:
@rossabaker @hamnis do you mind double checking if this is appropriate? 
I wasn't sure, because the middleware is supposed to be a complementary mechanism for combating CSRF.

Reply:
SameSite helps with CSRF, so I do think this is fine.